lib/src/main/java/org/altbeacon/beacon/service/ScanState.java

```diff
@@ -140,13 +146,26 @@ public static ScanState restore(Context context) {
             FileInputStream inputStream = null;
             ObjectInputStream objectInputStream = null;
             try {
-                inputStream = context.openFileInput(STATUS_PRESERVATION_FILE_NAME);
-                objectInputStream = new ObjectInputStream(inputStream);
-                scanState = (ScanState) objectInputStream.readObject();
-                scanState.mContext = context;
-            } catch (FileNotFoundException fnfe) {
+                File file = context.getFileStreamPath(STATUS_PRESERVATION_FILE_NAME);
+                if (file.length() > SANITY_FILE_SIZE_LIMIT) {
+                    // make sure file size is reasonable.  If over 100k, do not restore
+                    // See issue #1129
+                    LogManager.e(TAG, "Refusing to restore file of size "+file.length());
```

Comment:
Could we maybe delete `STATUS_PRESERVATION_FILE_NAME` if it is too big or corrupted?

Reply:
I think the code does something similar as is because as soon as the file is persisted (typically after app processing a few milliseconds later) the scan state is persisted with the newly emptier state, overwriting the problematic file.